Sources/main.swift

```diff
@@ -71,94 +56,73 @@ struct Entrypoint {
             let json: Input
         }
 
-        let httpClient = HTTPClient(
-            eventLoopGroupProvider: .singleton
+        var request = HTTPClientRequest(url: "https://httpbin.org/post")
+        request.method = .POST
+        request.headers.add(name: "content-type", value: "application/json")
+        
+        let input = Input(
+            id: 1,
+            title: "foo",
+            completed: false
         )
-        do {
-            var request = HTTPClientRequest(
-                url: "https://httpbin.org/post"
-            )
-            request.method = .POST
-            request.headers.add(name: "content-type", value: "application/json")
-            
-            let input = Input(
-                id: 1,
-                title: "foo",
-                completed: false
-            )
-            
-            let encoder = JSONEncoder()
-            let data = try encoder.encode(input)
-            let buffer = ByteBuffer(bytes: data)
-            request.body = .bytes(buffer)
-            
-            let response = try await httpClient.execute(
-                request,
-                timeout: .seconds(5)
-            )
-            
-            if response.status == .ok {
-                if let contentType = response.headers.first(
-                    name: "content-type"
-                ), contentType.contains("application/json") {
-                    var buffer: ByteBuffer = .init()
-                    for try await var chunk in response.body {
-                        buffer.writeBuffer(&chunk)
-                    }
-                    
-                    let decoder = JSONDecoder()
-                    if let data = buffer.getData(at: 0, length: buffer.readableBytes) {
-                        let output = try decoder.decode(Output.self, from: data)
-                        print(output.json.title)
-                    }
-                }
-
-            }
-            else {
-                print("Invalid status code: \(response.status)")
-            }
+        
+        let encoder = JSONEncoder()
+        let data = try encoder.encode(input)
+        request.body = .bytes(.init(bytes: data))
+        
+        let response = try await HTTPClient.shared.execute(
+            request,
+            timeout: .seconds(5)
+        )
+        
+        guard response.status == .ok else {
+            print("Invalid status code: \(response.status)")
+            return
+        }
+        guard
+            let contentType = response.headers.first(name: "content-type"),
+            contentType.contains("application/json")
+        else {
+            print("Invalid content type.")
+            return
         }
-        catch {
-            print("\(error)")
+
+        var buffer: ByteBuffer = .init()
+        for try await var chunk in response.body {
```

Comment:
This isn't even needed. You can just `.collect` the chunks into a single buffer with a NIO-provided helper on AsyncSequence